extension-manifest-v3/src/background/reporting.js

```diff
@@ -216,26 +216,43 @@ const reporting = new Reporting({
   communication,
 });
 
-observe('terms', (terms) => {
+function delay(timeInMs) {
+  return new Promise((resolve) => setTimeout(resolve, timeInMs));
+}
+
+function defer() {
+  const deferred = {};
+  deferred.promise = new Promise((resolve) => {
+    deferred.resolve = resolve;
+  });
+  return deferred;
+}
+
+let startupDefer = defer();
+
+observe('terms', async (terms) => {
+  if (!startupDefer) {
+    startupDefer = defer();
+  }
+
   if (terms) {
-    reporting.init().catch((e) => {
+    try {
+      await reporting.init();
+    } catch (e) {
       console.warn(
         'Failed to initialize reporting. Leaving the module disabled and continue.',
         e,
       );
-    });
+    }
   } else {
     reporting.unload();
   }
-});
 
-function delay(timeInMs) {
-  return new Promise((resolve) => setTimeout(resolve, timeInMs));
-}
+  startupDefer.resolve();
+  startupDefer = null;
```

Comment:
I wouldn't "null" it out. (And thus you can also simplify a bit by removing
```
 if (!startupDefer) {
    startupDefer = defer();
  }
  ```
  and later you don't have to check for being null.
  

The reason is that the semantic change between nulling and not nulling is that when terms are toggled (but it has been already initialized), it will again wait. Without nulling, it will not wait.

Not waiting is the preferred semantic here IMHO. And has also the benefit that its simplifies the implementation.